django/db/backends/base/schema.py

```diff
@@ -893,6 +906,24 @@ def _alter_column_type_sql(self, model, old_field, new_field, new_type):
             [],
         )
 
+    def _alter_column_collation_sql(self, new_field, new_type, new_collation):
+        if new_field.db_collation is None:
+            return (
+                self.sql_alter_column_no_collate % {
+                    'column': self.quote_name(new_field.column),
+                    'type': new_type,
+                },
+                [],
+            )
+        return (
+            self.sql_alter_column_collate % {
+                'column': self.quote_name(new_field.column),
+                'type': new_type,
+                'collation': self.quote_name(new_collation),
+            },
+            [],
+        )
```

Comment:
We can simplify this with changing a template:
```python
sql = self.sql_alter_column_collate if new_field.db_collation else self.sql_alter_column_no_collate
return sql % {
        'column': self.quote_name(new_field.column),
        'type': new_type,
        'collation': self.quote_name(new_collation),
    },
    [],
}
```

Reply:
I ran into errors trying this, e.g.:

```
======================================================================
ERROR: test_collations (schema.tests.SchemaTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/tom/work/personal/django/django/test/utils.py", line 386, in inner
    return func(*args, **kwargs)
  File "/Users/tom/work/personal/django/django/test/testcases.py", line 1289, in skip_wrapper
    return test_func(*args, **kwargs)
  File "/Users/tom/work/personal/django/tests/schema/tests.py", line 3217, in test_collations
    editor.alter_field(
  File "/Users/tom/work/personal/django/django/db/backends/base/schema.py", line 579, in alter_field
    self._alter_field(model, old_field, new_field, old_type, new_type,
  File "/Users/tom/work/personal/django/django/db/backends/base/schema.py", line 695, in _alter_field
    fragment = self._alter_column_collation_sql(new_field, new_type, new_collation)
  File "/Users/tom/work/personal/django/django/db/backends/base/schema.py", line 917, in _alter_column_collation_sql
    'collation': self.quote_name(new_collation),
  File "/Users/tom/work/personal/django/django/db/backends/base/schema.py", line 147, in quote_name
    return self.connection.ops.quote_name(name)
  File "/Users/tom/work/personal/django/django/db/backends/mysql/operations.py", line 172, in quote_name
    if name.startswith("`") and name.endswith("`"):
AttributeError: 'NoneType' object has no attribute 'startswith'
```

Reply:
Nevermind, got a variant of it working.